modules/forum/src/main/ForumDelete.scala

```diff
@@ -39,6 +39,13 @@ final class ForumDelete(
       .runWith(Sink.ignore)
       .void
 
+  def deleteTopic(view: PostView) =
+    for
+      _ <- postRepo removeByTopic view.topic.id
+      _ <- topicRepo.remove(view.topic)
+      _ <- categApi denormalize view.categ
+    yield indexer ! RemovePost(view.post.id)
```

Comment:
if `deleteTopic` is called with a post that's not the only one in its topic, then all the other posts of the topic will not be removed from the indexer.